services/sts/src/main/java/software/amazon/awssdk/services/sts/auth/StsGetFederationTokenCredentialsProvider.java

```diff
@@ -64,11 +66,23 @@ public static Builder builder() {
     }
 
     @Override
-    protected Credentials getUpdatedCredentials(StsClient stsClient) {
-        return stsClient.getFederationToken(getFederationTokenRequest).credentials();
+    protected SessionCredentialsHolder getUpdatedCredentials(StsClient stsClient) {
+        GetFederationTokenResponse federationToken = stsClient.getFederationToken(getFederationTokenRequest);
+        return SessionCredentialsHolder.builder()
+                                       .credentials(federationToken.credentials())
+                                       .accountId(accountIdFromArn(federationToken.federatedUser()))
+                                       .build();
+    }
+
+    private String accountIdFromArn(FederatedUser federatedUser) {
+        if (federatedUser == null) {
+            return null;
+        }
+        return Arn.parse(federatedUser.arn())
+                  .map(Arn::accountId)
+                  .orElse(null);
     }
 
-    @Override
```

Comment:
`@Override` can remain

Reply:
hm, yes that must've happened by mistake